### PR TITLE
 Fixed timedout + retry and added tests

### PR DESCRIPTION
**Pull request by davidstl**

### CODE REVIEW

File: BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/Internal/BrainCloudComms.cs [hunk 1]
```diff
@@ -421,8 +424,10 @@ public void Update()
             // is it time for a retry?
             if (_activeRequest != null)
             {
-                if (DateTime.Now.Subtract(_activeRequest.TimeSent) >= GetPacketTimeout(_activeRequest))
+                if (bypassTimeout || DateTime.Now.Subtract(_activeRequest.TimeSent) >= GetPacketTimeout(_activeRequest))
                 {
+                    _clientRef.Log(string.Format("Timeout passed: {0}",  DateTime.Now));
```
Comment (davidstl) [inline BrainCloudClient/Assets/BrainCloud/Client/BrainCloud/Internal/BrainCloudComms.cs]: oops will remove

**Comment by stevemrjones** [comment]
+1

File: BrainCloudClient/tests/TestComms.cs [hunk 2]
```diff
@@ -334,6 +334,17 @@ public void TestAuthFirst()
             tr.Run();
         }
 
+        [Test]
+        public void TestRetry30Sec()
```
Comment (davidstl) [inline BrainCloudClient/tests/TestComms.cs]: oops will remove this. Duplicated

File: BrainCloudClient/tests/TestFixtureBase.cs [hunk 3]
```diff
@@ -163,7 +163,7 @@ private void LoadIds()
         {
             string exePath = AppDomain.CurrentDomain.BaseDirectory;
             string absPath = exePath;
-            string search = "Unity-Csharp" + Path.DirectorySeparatorChar + "BrainCloudClient";
+            string search = "BrainCloudClient";
```
Comment (davidstl) [inline BrainCloudClient/tests/TestFixtureBase.cs]: Older code was assuming people clone it in always in the same folder name.

**Pull request merged**